src/engine/sidechain/engineshoutcast.cpp

```diff
@@ -533,13 +558,39 @@ bool EngineShoutcast::writeSingle(const unsigned char* data, size_t len) {
 
 void EngineShoutcast::process(const CSAMPLE* pBuffer, const int iBufferSize) {
 
-    setState(NETWORKSTREAMWORKER_STATE_BUSY);
     // If we are here then the user wants to be connected (shoutcast is enabled
     // in the preferences).
 
-    // If we aren't connected, bail.
-    if (m_iShoutStatus != SHOUTERR_CONNECTED)
+    // If we aren't connected, bail and autoreconnect is off.
+    if (m_iShoutStatus != SHOUTERR_CONNECTED
+        && !m_stream_autoreconnect) {
         return;
+    } else if (m_iShoutStatus != SHOUTERR_CONNECTED
+               && m_stream_autoreconnect
+               && getState() != NETWORKSTREAMWORKER_STATE_CONNECTING) {
+        bool l_bConnectionSuccess = false;
+        // Trying to connect to server
+        // If autoreconnect is on we just keep trying until
+        // the end of universe or Shoutcast is not enabled anymore
+        while (true) {
```

Comment:
We may also check for the "enabled" CO in the loop. This way we can continue very long, without delay the exit. 
The same for the other loop.

By the way, cam one confirm the deadlock or do we hunt a phantom?